CCPAPI/PAPI/ccPApi.cpp

```diff
@@ -0,0 +1,476 @@
+//##########################################################################
+//#                                                                        #
+//#                               CCPAPI                                   #
+//#                                                                        #
+//#  This program is free software; you can redistribute it and/or modify  #
+//#  it under the terms of the GNU Library General Public License as       #
+//#  published by the Free Software Foundation; version 2 or later of the  #
+//#  License.                                                              #
+//#                                                                        #
+//#  This program is distributed in the hope that it will be useful,       #
+//#  but WITHOUT ANY WARRANTY; without even the implied warranty of        #
+//#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the          #
+//#  GNU General Public License for more details.                          #
+//#                                                                        #
+//#          Copyright 2020 Paul RASCLE www.openfields.fr                  #
+//#                                                                        #
+//##########################################################################
+
+#include "ccPApi.h"
+
+#include <GeometricalAnalysisTools.h>
+
+//qCC_db
+#include <ccHObjectCaster.h>
+
+//libs/qCC_io
+#include<AsciiFilter.h>
+
+//system
+#include <unordered_set>
+
+//plugins
+#include "ccPluginInterface.h"
+#include "ccPluginManager.h"
+
+//qCC
+#include "ccCommon.h"
+
+#include "ccTrace.h"
+
+using namespace PAPI;
```

Comment:
Suggestion: wrap the things that are part of PAPI in

```cpp
namespace PAPI {
 <stuff in PAPI namespace>
}
```

It's more explicit and can avoid name clashes.

So (based on the comment that follows) the structure of the file is:

```cpp
namespace {

 struct CLLoadParameters: public FileIOFilter::LoadParameters

 // ...<any other things that are only available to this compilation unit>

}

namespace PAPI {
 // <stuff in PAPI namespace>
}
```